### PR TITLE
Fix Actions calling next action multiple times

### PR DESCRIPTION
**Pull request by tlfn**
**Please describe the changes this PR makes and why it should be merged:**
Refactor the code
Fix calling next action multiple times

**Status**

- [x] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [x] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by tlfn** [comment]
I see no reason to reject printing something in Send Message to Console

**Comment by Lucasamiel0406** [comment]
Tested it on my dead DBM bot, working as intended now.

**Comment by Lucasamiel0406** [comment]
And `Send Message to Console` also works, both tested in latest version of Mods.

**Review by Lucasamiel0406 — APPROVED** [review]

**Review by TheMonDon — APPROVED** [review]

File: actions/set_bot_activity_MOD.js [hunk 1]
```diff
@@ -101,92 +101,28 @@ module.exports = {
...
-        case 3:
-          statustarget = 'dnd';
-          break;
-        default:
-          break;
-      }
+    if (!botClient) {
+      console.error('ERROR: Please input activity in "Set Bot Activity MOD"');
+      return this.callNextAction(cache);
     }
```
Comment (almeidx) [inline actions/set_bot_activity_MOD.js:110]:
```suggestion
```

File: actions/set_bot_activity_MOD.js [hunk 2]
```diff
@@ -101,92 +101,28 @@ module.exports = {
     const nameText = this.evalMessage(data.nameText, cache);
```
Comment (almeidx) [inline actions/set_bot_activity_MOD.js:101]:
```suggestion
    const nameText = this.evalMessage(data.nameText, cache) || null;
```

File: actions/set_bot_activity_MOD.js [hunk 3]
```diff
@@ -101,92 +101,28 @@ module.exports = {
...
-          botClient
-            .setPresence(obj)
-            .then(() => {
-              this.callNextAction(cache);
-            })
-            .catch(console.error);
-        }
-      } else {
-        console.log('ERROR: Please input activity in "Set Bot Activity MOD"');
+    if (!nameText) return this.callNextAction(cache);
```
Comment (almeidx) [inline actions/set_bot_activity_MOD.js:111]:
```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: actions/set_bot_activity_MOD.js [hunk 4]
```diff
@@ -98,95 +98,26 @@ module.exports = {
...
+        name: nameText,
+        type: target,
+      },
+      status: statusTarget,
+    };
+    if (target === 'STREAMING') Object.assign(obj.activity, { url });
+    botClient
+      .setPresence(obj)
+      .then(() => {
         this.callNextAction(cache);
```
Comment (almeidx) [inline actions/set_bot_activity_MOD.js:118]: Can't comment on deleted lines, but make this inline `() => this.callNextAction(cache)`

Comment (tlfn) [reply actions/set_bot_activity_MOD.js]: Reflected in the latest commit

File: actions/set_bot_activity_MOD.js [hunk 5]
```diff
@@ -98,95 +98,24 @@ module.exports = {
...
+        name: nameText,
+        type: target,
+      },
+      status: statusTarget,
+    };
+    if (target === 'STREAMING') Object.assign(obj.activity, { url });
+    botClient
+      .setPresence(obj)
+      .then(() => this.callNextAction(cache))
+      .catch(console.error);
```
Comment (almeidx) [inline actions/set_bot_activity_MOD.js:118]:
```suggestion
      .catch(this.displayError.bind(this, data, cache));
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request merged**